### ISSUE TITLE
I don't understand why the Loss_D should like that.

### ISSUE THREAD

**Issue by Huixxi** [sciencefans/D2AE-Face-Generator#2]
When I tried to implement the front part of your D2AE network architecture involving the Encode, two branch sunnets, and I also tried to train that part just to get the identity-distilling feature of the face images, but I found the loss_D seems to large to converge, besides I don't understand why loss d should be the negative entropy, what the function of that loss?

**Comment by mingzhang96** [comment]
loss_d should be negative as it is used to fool the identity classifier so that the classifier will get the feature of some information not correspond with identity such as background noise, opening mouth, smiling and so on. The author wants this loss to be bigger so that he add a negative sign.

By the way, how do you do the backward propagation? I have some questions on it. I don't know how to do backward propagation for Encoder and two subnets with classifier fixed while doing another propagation on classifier in one batch. I use two optimizer and backward them separately. It may be not correct and use too much memory. And I also raise this question [here](https://discuss.pytorch.org/t/how-to-update-sequential-part-of-model-using-different-loss/40838/2). Could you help me solve this problem? Thanks!

**Comment by Dwrety** [comment]
@mingzhang96 I used two optimizers, in PyTorch, this would look like:
`    optimizer_all = optim.SGD([{'params':d2ae.encoder.parameters()},
                               {'params':d2ae.decoder.parameters()},
                               {'params':d2ae.subnet_t.parameters()},
                               {'params':d2ae.subnet_p.parameters()},
                               {'params':d2ae.idt.parameters()}], lr=0.002, momentum=0.9)`
`    optimizer_adv = optim.SGD([{'params':d2ae.idp.parameters()}], lr=0.002, momentum=0.9)`
But I haven't been able to see convergence yet! 
Can you elaborate a little bit on the adversarial LI_adv and L_H please. It was described in paper as the negative entropy of y_p, but the equation given is the sum of log(y_p) not y_p*log(y_p). I don't understand what you are minimizing here? Sum of log(y_p)? This could take a value of -INF, right? I understand that finally the distribution should be Uniform, but what about the early stage?
